### PR TITLE
Fix to provide a way to retrieve user fields using graph v2.4

### PR DESCRIPTION
**Pull request by jsenin**

### CODE REVIEW

File: FacebookStrategy.php [hunk 1]
```diff
@@ -118,14 +126,17 @@ public function int_callback(){
...
 	 * Queries Facebook Graph API for user info
 	 *
 	 * @param string $access_token 
 	 * @return array Parsed JSON results
 	 */
-	private function me($access_token){
-		$me = $this->serverGet('https://graph.facebook.com/me', array('access_token' => $access_token), null, $headers);
+	private function me($access_token, $fields = array() ){
+
+		$me = $this->serverGet('https://graph.facebook.com/me', array('access_token' => $access_token) + compact('fields') , null, $headers);
```
Comment (Kryniol) [inline FacebookStrategy.php]: this won't work, fields should be comma-separated
I suggest to use implode

File: FacebookStrategy.php [hunk 2]
```diff
@@ -63,7 +63,12 @@ public function int_callback(){
 			parse_str($response, $results);
 
 			if (!empty($results) && !empty($results['access_token'])){
-				$me = $this->me($results['access_token']);
+				
+				$fields = array();
```
Comment (Kryniol) [inline FacebookStrategy.php]: @jsenin I was working on the same issue, spotted yours when preparing a PR :)
my solution was to query for all the fields which are used then in int_callback() method (email, first_name, last_name, location, link and website), this wouldn't require any development for existing applications and attempting to retrieve a field which an app doesn't have access to doesn't trigger any errors

Comment (jsenin) [inline FacebookStrategy.php]: @Kryniol , the idea is configure 'fields' as settings, so you can set comma separated fields like this :

<?php
'Facebook' => array(
    'app_id' => 'YOUR APP ID',
    'app_secret' => 'YOUR APP SECRET'
    'scope' => 'email',
    'fields' => 'id,first_name,last_name,gender,email', 
)
I had write a little aditional info at README
https://github.com/jsenin/facebook/blob/master/README.md

It's the minor change that i need to keep it working.

greets

Comment (Kryniol) [inline FacebookStrategy.php]: @jsenin me too :)
Did you check https://github.com/opauth/facebook/pull/26?

**Comment by jsenin** [comment]
I just to see the #26 and #21 and I see that fields are now hardcoded :(

**Comment by Kryniol** [comment]
@jsenin because majority of them is being used by the strategy class itself. Maybe the best solution would be to have the used ones hardcoded and allow to configure what else should be retrieved. On the other hand querying for unavailable fields won't throw any exception (they just won't be returned) so for me the solution with all hardcoded fields is acceptable.

**Comment by jsenin** [comment]
I see, it's ok for me too :D

2015-09-09 14:44 GMT+02:00 Paweł Krynicki <notifications@github.com>:

**Pull request closed without merging**